src/osemgrep/reporting/Status_report.ml

```diff
@@ -27,13 +27,24 @@ let pp_status ~num_rules ~num_targets ~respect_git_ignore lang_jobs ppf =
   *)
   Fmt.pf ppf ":@.@.";
   (* TODO origin table [Origin Rules] [Community N] *)
+  let lan_label = function
+    | Xlang.LGeneric
+    | Xlang.LRegex ->
+        "<multilang>"
+    | lang -> Xlang.to_string lang
+  in
   Fmt_helpers.pp_table
     ("Language", [ "Rules"; "Files" ])
     ppf
     (lang_jobs
+    |> Common.map (fun Lang_job.{ lang; targets; rules } ->
+           (lan_label lang, List.length rules, List.length targets))
     |> List.fold_left
-         (fun acc Lang_job.{ lang; targets; rules } ->
-           (Xlang.to_string lang, [ List.length rules; List.length targets ])
-           :: acc)
+         (fun acc (lang, rules, targets) ->
+           match List.partition (fun (l, _) -> l = lang) acc with
+           | [], others -> (lang, [ rules; targets ]) :: others
+           | [ (_, [ r1; t1 ]) ], others ->
+               (lang, [ rules + r1; targets + t1 ]) :: others
+           | _ -> assert false)
```

Comment:
this ..map.. fold_left looks ugly, and likely the Common.group_by has a better complexity

Reply:
yes ... when I fold and fold and fold and I need to thread things, sometimes I rewrite the code imperatively
and use a let res = ref [] in. and use List.iter and iter and iter, and update the res.
It's sometimes cleaner.